rest/taskana-rest-spring/src/main/java/pro/taskana/task/rest/TaskController.java

```diff
@@ -544,17 +547,75 @@ public ResponseEntity<TaskRepresentationModel> terminateTask(@PathVariable Strin
   public ResponseEntity<TaskRepresentationModel> transferTask(
       @PathVariable String taskId,
       @PathVariable String workbasketId,
-      @RequestBody(required = false) Boolean setTransferFlag)
+      @RequestBody(required = false)
+          TransferTaskRepresentationModel transferTaskRepresentationModel)
       throws TaskNotFoundException,
           WorkbasketNotFoundException,
           NotAuthorizedOnWorkbasketException,
           InvalidTaskStateException {
-    Task updatedTask =
-        taskService.transfer(taskId, workbasketId, setTransferFlag == null || setTransferFlag);
+    Task updatedTask;
+    if (transferTaskRepresentationModel == null) {
+      updatedTask = taskService.transfer(taskId, workbasketId);
+    } else {
+      updatedTask =
+          taskService.transferWithOwner(
+              taskId,
+              workbasketId,
+              transferTaskRepresentationModel.getOwner(),
+              transferTaskRepresentationModel.getSetTransferFlag());
+    }
 
     return ResponseEntity.ok(taskRepresentationModelAssembler.toModel(updatedTask));
   }
 
+  /**
+   * This endpoint transfers a list of Tasks listed in the body to a given Workbasket, if possible.
```

Comment:
Can you decsribe the behaviour more? For example, it's important that each Task that can be tranferred without an exception gets transferred independent of other Tasks. It should be also mentioned what happens if a Task can't be transferred

Reply:
I have described it more, please check it :)